### PR TITLE
Adding Email to student features

### PR DESCRIPTION
**Pull request by wyntonfranklin**
Adding Email to student features

### CODE REVIEW

File: app/Http/Controllers/EmailController.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Mail;
+
+class EmailController extends Controller
+{
+
+    public function __construct()
+    {
+        parent::__construct();
+      //  $this->middleware('permission:hr.view', ['except' => 'teacher']);
```
Comment (thdebay) [inline app/Http/Controllers/EmailController.php:16]: Here the middleware should be `permission:student.edit`

File: app/Http/Controllers/EmailController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(Request $request)
+    {
+        $student = "";
+        if($request->get("student")){
+            $student = $request->get("student");
+        }
```
Comment (thdebay) [inline app/Http/Controllers/EmailController.php:27]: This seems a little odd. Why do you assign an empty string to the $student variable ? I think you could add an early return here (if request does not have "student", abort and redirect and flash an error message) and then on line 34, do: `'preselect' => $request->get("student"),`

File: app/Http/Controllers/EmailController.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        if($email == "all"){
+            $email = $this->getEmails($courseId);
+        }
+
+        Mail::to($email)->send(new EmailToStudent($subject, $message));
+        // todo: redirect not working for some reason
+        return redirect(route('email-dashboard'))->with('success', 'Profile updated!');
+    }
+
+    private function getEmails($courseId){
```
Comment (thdebay) [inline app/Http/Controllers/EmailController.php:61]:
```suggestion
    private function getEmails(int $courseId){
```

**Comment by wyntonfranklin** [comment]
@thdebay will make the suggested changes.

File: app/Http/Controllers/EmailController.php (hunk 2 above)
Comment (wyntonfranklin) [reply app/Http/Controllers/EmailController.php:27]: where would I be redirecting to?

**Comment by wyntonfranklin** [comment]
thanks :) I think its ready

**Comment by thdebay** [comment]
Perfect, thank you again for your work!! This is a great addition to the project.

I merged it "as-is" for now, we can see how it works! A potential security drawback I see is that any teacher will be able to send an email to any student - not just their students. But I think we can leave it like this for now, maybe it will be a topic for another improvement in the future!

**Comment by thdebay** [comment]
@all-contributors please add @wyntonfranklin for code

**Pull request merged**